code/iaas/cluster-logic/src/main/java/io/cattle/iaas/cluster/process/ClusterHostMapCreateRemoveHandler.java

```diff
@@ -0,0 +1,41 @@
+package io.cattle.iaas.cluster.process;
+
+import io.cattle.iaas.cluster.service.ClusterManager;
+import io.cattle.platform.core.model.ClusterHostMap;
+import io.cattle.platform.core.model.Host;
+import io.cattle.platform.engine.handler.HandlerResult;
+import io.cattle.platform.engine.handler.ProcessPostListener;
+import io.cattle.platform.engine.process.ProcessInstance;
+import io.cattle.platform.engine.process.ProcessState;
+import io.cattle.platform.process.common.handler.AbstractObjectProcessLogic;
+import io.cattle.platform.util.type.Priority;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+@Named
+public class ClusterHostMapCreateRemoveHandler extends AbstractObjectProcessLogic implements ProcessPostListener, Priority {
+
+    @Inject
+    ClusterManager clusterManager;
+
+    @Override
+    public String[] getProcessNames() {
+        return new String[] { "clusterhostmap.create", "clusterhostmap.remove" };
+    }
+
+    @Override
+    public HandlerResult handle(ProcessState state, ProcessInstance process) {
+        Long clusterId = ((ClusterHostMap) state.getResource()).getClusterId();
+        Host cluster = objectManager.loadResource(Host.class, clusterId);
```

Comment:
I see a potential problem here for the clusterhostmap.remove. If the host that is being removed is a master host, the updated config won't have this host specified. So you will end up updating the master with the config not having the master. Unless master host is reassigned by this point. I think we should add a pre-handler for clusterhostmap.remove checking if the host-to-remove is a master, then reassigning an another master, and only then updating the config. But you can do it as part of a PR that is gonna take care of all the cleanup/management reassignment cases. By now, just make sure that config update lacking master, doesn't result in the error on the backend.